### PR TITLE
v1 open api refactor

### PR DESCRIPTION
**Pull request by JamesCamel**

### CODE REVIEW

File: nodejs/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended"
+    ],
+    "rules": {
+        "no-trailing-spaces": ["warn", { "skipBlankLines": false }],
+        "semi": ["warn", "never"],
+        "@typescript-eslint/no-non-null-assertion": "off",
+        "keyword-spacing": ["warn", {"before": true, "after": true}],
+        "space-infix-ops": "warn",
+        "space-before-function-paren": "warn"
```
Comment (Yukaii) [inline nodejs/.eslintrc.js:15]: Also turn on the eol rule: https://eslint.org/docs/rules/eol-last

`"eol-last": ["error", "always"]`

File: nodejs/package.json [hunk 2]
```diff
@@ -21,22 +22,17 @@
...
-    "cheerio": "^1.0.0-rc.3",
-    "fetch-cookie": "^0.8.0",
-    "fs-extra": "^9.0.0",
-    "node-fetch": "^2.6.0",
-    "tough-cookie": "^3.0.0",
-    "tough-cookie-filestore": "^0.0.1",
+    "@typescript-eslint/eslint-plugin": "^5.12.0",
+    "axios": "^0.25.0",
+    "path": "^0.12.7",
     "tslib": "^1.11.1"
```
Comment (Yukaii) [inline nodejs/package.json:36]: we can bump `tslib` version as well

Comment (Yukaii) [reply nodejs/package.json:36]: And made our package version to 2.0.0 from `"version": "1.2.0",`

**Review by Yukaii — CHANGES_REQUESTED** [review]

File: nodejs/.eslintrc.js (hunk 1 above)
Comment (JamesCamel) [reply nodejs/.eslintrc.js]: Added.

File: nodejs/package.json (hunk 2 above)
Comment (JamesCamel) [reply nodejs/package.json]: Fixed

File: nodejs/src/index.ts [hunk 3]
```diff
@@ -1,262 +1,111 @@
...
+  async updateTeamNoteContent (teamPath: string, noteId: string, content?: string): Promise<UpdateTeamNote> {
+    await this.axios.patch<AxiosResponse>(`teams/${teamPath}/notes/${noteId}`, { content })
   }
-}
 
-export default API
+  async deleteTeamNote (teamPath: string, noteId: string): Promise<DeleteTeamNote> {
+    await this.axios.delete<AxiosResponse>(`teams/${teamPath}/notes/${noteId}`)
+  }
+}
```
Comment (Yukaii) [inline nodejs/src/index.ts:111]: EOL should be added (with the latest linter config)

Comment (JamesCamel) [reply nodejs/src/index.ts]: Added.

**Review by Yukaii — APPROVED** [review]

**Pull request merged**